include/rmm/mr/device/detail/arena.hpp

```diff
@@ -345,6 +351,38 @@ class global_arena final {
     }
   }
 
+  /**
+   * @brief Dump memory to log.
+   *
+   * @param logger the spdlog logger to use
+   */
+  void dump_memory_log(std::shared_ptr<spdlog::logger> const& logger) const
+  {
+    lock_guard lock(mtx_);
+
+    logger->info("  Maximum size: {}", rmm::detail::bytes{maximum_size_});
+    logger->info("  Current size: {}", rmm::detail::bytes{current_size_});
+
+    logger->info("  # free blocks: {}", free_blocks_.size());
+    if (!free_blocks_.empty()) {
+      std::size_t total_free{};
+      for (auto const& b : free_blocks_) {
+        total_free += b.size();
+      }
```

Comment:
```suggestion
  auto const total_free = std::accumulate(free_blocks_.cbegin(),  //
                                          free_blocks_.cend(),
                                          std::size_t{0},
                                          [](auto a, auto b) { return a + b.size(); });
```

In C++23, it will be really nice because we can do this:
```cpp
auto const total_free = std::accumulate(free_blocks_, std::size_t{0}, std::plus{}, &block::size);
```